### ISSUE TITLE
GRBL for dot peen engraver

### ISSUE THREAD

**Issue by badidea** [grbl/grbl#194]
I am working on retrofitting a dot peen engraver with GRBL and Arduino UNO.  I am having an issue I am sure you guys can help me with.  I am trying to put a spindle control  that when the steppers move a number of pulses it fires my spindle enable output for 10ms then lets off.  I use M3 as the start of the peen cycle. I have my Sxxx code that is put in with M3 as my dots spacing variable.   I can get all of it except the pulse counter working.  

I am new to programming these and am sure that I am missing something obvious.  I appreciate any help you can give.

Basically I just need to know how I can put in a counter that watches X and Y and resets when my output fires or M5 is put in.
